app/workers/set_tenant_id_worker.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class SetTenantIdWorker < ApplicationJob
+
+  def perform(provider_batch)
+    provider_batch.each do |provider|
+      # Set the master field to false
+      provider.master = false
+      provider.save!
+
+      # Update the backend_apis, log_entries and alerts to propagate the tenant_id from the updated
+      # provider
+      {
+        BackendApi => :account_id,
+        LogEntry => :provider_id,
+        Alert => :account_id,
+      }.each do |model, relation|
+        q = {}
+        q[relation] = provider.id
```

Comment:
If `q` this doesn't depend on model/relation, can we move outside of the loop?
Also, maybe
```suggestion
        q = Hash(relation, provider.id)
```